add docs for supabase CLI completion (#9589)
supabase/supabase: apps/reference/_cli/intro.mdx

- CI/CD for releasing to production: [`supabase db push`](https://supabase.com/docs/reference/cli/usage#supabase-db-push)
- Manage your Supabase projects: [`supabase projects`](https://supabase.com/docs/reference/cli/usage#supabase-projects)
- Generate types directly from your database schema: [`supabase gen types`](https://supabase.com/docs/reference/cli/usage#supabase-gen)
    - A [community-supported GitHub Action to generate Supabase DB types](https://github.com/lyqht/generate-supabase-db-types-github-action) also uses this command
- Shell Autocomplete: [`supabase completion`](https://supabase.com/docs/reference/cli/usage#supabase-completion)

>     - A community Fig autocomplete spec for supabase is also available

Author: added a link!

>     - A [community-supported GitHub Action to generate TypeScript types](https://github.com/lyqht/generate-supabase-db-types-github-action) using this command
> - Shell autocomplete: [`supabase completion`](https://supabase.com/docs/reference/cli/usage#supabase-completion)
>     - A [community-supported Fig autocomplete spec for Supabase](https://fig.io/manual/supabase) is also available

Reviewer: ```suggestion
    - A community-supported [GitHub Action to generate TypeScript types](https://github.com/lyqht/generate-supabase-db-types-github-action)
- Shell autocomplete: [`supabase completion`](https://supabase.com/docs/reference/cli/usage#supabase-completion)
    - A community-supported [Fig autocomplete spec for macOS terminal](https://fig.io/manual/supabase)
```
